redbot/cogs/cleanup/cleanup.py

```diff
@@ -1,345 +1,345 @@
+    async def text(self, ctx: commands.Context, text: str, number: int):
+        """Deletes last X messages matching the specified text.
+
+        Example:
+        cleanup text \"test\" 5
+
+        Remember to use double quotes."""
+
+        channel = ctx.channel
+        author = ctx.author
+        is_bot = self.bot.user.bot
+
+        def check(m):
+            if text in m.content:
+                return True
+            elif m == ctx.message:
+                return True
+            else:
+                return False
+
+        to_delete = [ctx.message]
+        too_old = False
+        tmp = ctx.message
+
+        while not too_old and len(to_delete) - 1 < number:
+            async for message in channel.history(limit=1000,
+                                                 before=tmp):
+                if len(to_delete) - 1 < number and check(message) and\
+                        (ctx.message.created_at - message.created_at).days < 14:
+                    to_delete.append(message)
+                elif len(to_delete) == number or (ctx.message.created_at - message.created_at).days >= 14:
+                    too_old = True
+                    break
+                tmp = message
+
+        reason = "{}({}) deleted {} messages "\
+                 " containing '{}' in channel {}".format(author.name,
+                                                         author.id, len(to_delete), text, channel.id)
+        log.info(reason)
+
+        if is_bot:
+            await mass_purge(to_delete, channel)
+        else:
+            await slow_deletion(to_delete)
+
+    @cleanup.command()
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_messages=True)
+    async def user(self, ctx: commands.Context, user: discord.Member or int, number: int):
+        """Deletes last X messages from specified user.
+
+        Examples:
+        cleanup user @\u200bTwentysix 2
+        cleanup user Red 6"""
+
+        channel = ctx.channel
+        author = ctx.author
+        is_bot = self.bot.user.bot
+
+        def check(m):
+            if isinstance(user, discord.Member) and m.author == user:
+                return True
+            elif m.author.id == user:  # Allow finding messages based on an ID
+                return True
+            elif m == ctx.message:
+                return True
+            else:
+                return False
+
+        to_delete = []
+        too_old = False
+        tmp = ctx.message
+
+        while not too_old and len(to_delete) - 1 < number:
+            async for message in channel.history(limit=1000,
+                                                 before=tmp):
+                if len(to_delete) - 1 < number and check(message) and\
+                        (ctx.message.created_at - message.created_at).days < 14:
+                    to_delete.append(message)
+                elif len(to_delete) == number or (ctx.message.created_at - message.created_at).days >= 14:

```

Comment:
Add another elif clause that doesn't set `too_old = True` and uses `len(to_delete) >= number` to be safe.